File: python/tempCodeRunnerFile.py
class Solution:
    def productOfSelfExceptSelf(self, nums):
        # answer = [1] * (len(nums))
        # prefix = 1
        answer = [1] * (len(nums))
        print(answer)

        for i in range(1, len(nums)):
            answer[i] = answer[i - 1] * nums[i-1]
            print(i)

        print(f"{answer} after first loop")

        postfix = 1
        for j in range(len(nums) - 1, -1, -1):
            answer[j] *= postfix
            postfix *= nums[j]

        return answer


sol = Solution()
print(sol.productOfSelfExceptSelf([1, 2, 3, 4]))

File: cpp/tempCodeRunnerFile.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution
{
public:
    void insertionSort(vector<int> &arr)
    {
        for (int i = 1; i < arr.size(); i++)
        {
            int j = i;
            while (j > 0 && arr[j - 1] > arr[j])
            {
                int temp = arr[j];
                arr[j] = arr[j - 1];
                arr[j - 1] = temp;
                j--;
            }
        }

        cout << "After using insertion sort" << endl;

        for (int i = 0; i < arr.size(); i++)
        {
            cout << arr[i] << "  ";
        }
        cout << endl;
    }
};

int main()
{
    vector<int> nums = {5, 4, 3, 2, 1};
    Solution sol;
    for (int i = 0; i < nums.size(); i++)
    {
        cout << nums[i] << " ";
    }
    sol.insertionSort(nums);
}